"No Content" v2 Image component breaks AEM as a Content Service .model.json output

## Bug Report

**Current Behavior**
An unconfigured Image v2 Component on a page hierarchy serialized by AEM as a Content Service (`.model.json`) results in an unhandled exception breaking the JSON response (HTTP 500; empty response)

```
aused by: com.fasterxml.jackson.databind.JsonMappingException: (was java.lang.NullPointerException) (through reference chain: com.adobe.aem.guides.wkndevents.core.models.impl.HierarchyPageImpl[":items"]->java.util.LinkedHashMap["image"]->com.adobe.cq.wcm.core.components.internal.models.v2.ImageImpl["areas"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:394) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:353) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow(StdSerializer.java:316) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:729) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:166) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeFields(MapSerializer.java:722) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:643) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize(MapSerializer.java:33) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:727) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:721) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:166) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:2795) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at org.apache.sling.models.jacksonexporter.impl.JacksonExporter.export(JacksonExporter.java:117) [org.apache.sling.models.jacksonexporter:1.0.8]
	... 168 common frames omitted
Caused by: java.lang.NullPointerException: null
	at java.base/java.util.Collections$UnmodifiableCollection.<init>(Collections.java:1030)
	at java.base/java.util.Collections$UnmodifiableList.<init>(Collections.java:1303)
	at java.base/java.util.Collections.unmodifiableList(Collections.java:1290)
	at com.adobe.cq.wcm.core.components.internal.models.v2.ImageImpl.getAreas(ImageImpl.java:145) [com.adobe.cq.core.wcm.components.core:2.8.0]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:688) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:721) [com.fasterxml.jackson.core.jackson-databind:2.10.0]
	... 179 common frames omitted
```


**Expected behavior/code**
I expect unconfigured Core Component to to cause HTTP 500 errors when requested using AEM Content Services.

**Environment**
AEM version and patch level: AEM as a Cloud Service
Core Components version: 2.8.0
JRE version: Java 8

**Possible Solution**
The issue appears to happen because the `areas` field is never set when the Image Component does not "have content", however when AEM Content Services serializes the model to JSON it calls getter for it, which tried to create am unmodifiable list of a null object.

Generally, I think there is an anti-pattern in how Core Components tend to lump all logic into the @PostConstruct method. Ideally, getters handle getting/resolving their own data (ie. separation of concerns).  For example a nice pattern for Sling Model getters is:

```
List<String> areas;

public List<String> getAreas() {
   if (areas == null) {
        areas = new ArrayList<>();
        areas.add("area 51");
   } 

   return Collections.unmodifiableList(areas);
}
```

Using this pattern you can still "cache" the value of `areas` in the getter is called multiple times, but it also separates the logic for areas out clearly, so you can easily understand how/when the variable is instantiated, avoiding issues like this. It also is "lazy" in that the logic for to "get" the areas is only invoked if the areas are needed; if they aren't, then AEM doesn't have to do extra, unnecessary work.

--- a/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java
+++ b/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java
@@ -119,8 +119,6 @@ public class ImageImpl extends com.adobe.cq.wcm.core.components.internal.models.
                 srcUriTemplate += "?" + CONTENT_POLICY_DELEGATE_PATH + "=" + policyDelegatePath;
                 src += "?" + CONTENT_POLICY_DELEGATE_PATH + "=" + policyDelegatePath;
             }
-
-            buildAreas();
             buildJson();
         }
     }
@@ -142,48 +140,51 @@ public class ImageImpl extends com.adobe.cq.wcm.core.components.internal.models.
     }
 
     @Override
-    public List<ImageArea> getAreas() { return Collections.unmodifiableList(areas); }
+    public List<ImageArea> getAreas() {
+        if (areas == null ) {
+            areas = new ArrayList<>();
+            if (hasContent) {
+                String mapProperty = properties.get(Image.PN_MAP, String.class);
+                if (StringUtils.isNotEmpty(mapProperty)) {
+                    // Parse the image map areas as defined at {@code Image.PN_MAP}
+                    String[] mapAreas = StringUtils.split(mapProperty, "][");
+                    for (String area : mapAreas) {
+                        int coordinatesEndIndex = area.indexOf(')');
+                        if (coordinatesEndIndex < 0) {
+                            break;
+                        }
+                        String shapeAndCoords = StringUtils.substring(area, 0, coordinatesEndIndex + 1);
+                        String shape = StringUtils.substringBefore(shapeAndCoords, "(");
+                        String coordinates = StringUtils.substringBetween(shapeAndCoords, "(", ")");
+                        String remaining = StringUtils.substring(area, coordinatesEndIndex + 1);
+                        String[] remainingTokens = StringUtils.split(remaining, "|");
+                        if (StringUtils.isBlank(shape) || StringUtils.isBlank(coordinates)) {
+                            break;
+                        }
+                        if (remainingTokens.length > 0) {
+                            String href = StringUtils.removeAll(remainingTokens[0], "\"");
+                            if (StringUtils.isBlank(href)) {
+                                break;
+                            }
+                            String target = remainingTokens.length > 1 ? StringUtils.removeAll(remainingTokens[1], "\"") : "";
+                            String alt = remainingTokens.length > 2 ? StringUtils.removeAll(remainingTokens[2], "\"") : "";
+                            String relativeCoordinates = remainingTokens.length > 3 ? remainingTokens[3] : "";
+                            relativeCoordinates = StringUtils.substringBetween(relativeCoordinates, "(", ")");
+                            if (href.startsWith("/")) {
+                                href = Utils.getURL(request, pageManager, href);
+                            }
+                            areas.add(new ImageAreaImpl(shape, coordinates, relativeCoordinates, href, target, alt));
+                        }
+                    }
+                }
+            }
+        }
+        return Collections.unmodifiableList(areas);
+    }
 
     @Override
     public String getUuid() {
         return uuid;
-    }
-
-    protected void buildAreas() {
-        areas = new ArrayList<>();
-        String mapProperty = properties.get(Image.PN_MAP, String.class);
-        if (StringUtils.isNotEmpty(mapProperty)) {
-            // Parse the image map areas as defined at {@code Image.PN_MAP}
-            String[] mapAreas = StringUtils.split(mapProperty, "][");
-            for (String area : mapAreas) {
-                int coordinatesEndIndex = area.indexOf(')');
-                if (coordinatesEndIndex < 0) {
-                    break;
-                }
-                String shapeAndCoords = StringUtils.substring(area, 0, coordinatesEndIndex + 1);
-                String shape = StringUtils.substringBefore(shapeAndCoords, "(");
-                String coordinates = StringUtils.substringBetween(shapeAndCoords, "(", ")");
-                String remaining = StringUtils.substring(area, coordinatesEndIndex + 1);
-                String[] remainingTokens = StringUtils.split(remaining, "|");
-                if (StringUtils.isBlank(shape) || StringUtils.isBlank(coordinates)) {
-                    break;
-                }
-                if (remainingTokens.length > 0) {
-                    String href = StringUtils.removeAll(remainingTokens[0], "\"");
-                    if (StringUtils.isBlank(href)) {
-                        break;
-                    }
-                    String target = remainingTokens.length > 1 ? StringUtils.removeAll(remainingTokens[1], "\"") : "";
-                    String alt = remainingTokens.length > 2 ? StringUtils.removeAll(remainingTokens[2], "\"") : "";
-                    String relativeCoordinates = remainingTokens.length > 3 ? remainingTokens[3] : "";
-                    relativeCoordinates = StringUtils.substringBetween(relativeCoordinates, "(", ")");
-                    if (href.startsWith("/")) {
-                        href = Utils.getURL(request, pageManager, href);
-                    }
-                    areas.add(new ImageAreaImpl(shape, coordinates, relativeCoordinates, href, target, alt));
-                }
-            }
-        }
     }
 
 }

